cookbooks/kinetic_driving_force/kinetic-driving-force.cc

```diff
@@ -0,0 +1,545 @@
+            "./",
+            Patterns::DirectoryName(),
+            "The name of a directory where the model data is found. This path "
+            "may either be absolute (if starting with '/') or relative to "
+            "the current directory.");
+          prm.declare_entry("Data file name",
+                            "thermo_data.txt",
+                            Patterns::Anything(),
+                            "The file name of the model data.");
+
+          // Rheological parameters
+          prm.declare_entry("Viscosity",
+                            "1e21",
+                            Patterns::Double(0.),
+                            "Viscosity. Units: Pa s");
+          prm.declare_entry(
+            "Thermal viscosity exponent",
+            "0.0",
+            Patterns::Double(0.),
+            "The temperature dependence of viscosity. Dimensionless exponent.");
+          prm.declare_entry(
+            "Transition depths",
+            "150e3, 410e3, 660e3",
+            Patterns::List(Patterns::Double(0.)),
+            "A list of depths where the viscosity changes. Values must "
+            "monotonically increase. "
+            "Units: m");
+          prm.declare_entry(
+            "Viscosity prefactors",
+            "10.0, 0.1, 1.0, 10.0",
+            Patterns::List(Patterns::Double(0.)),
+            "A list of prefactors for the viscosity that determine the "
+            "viscosity profile. Each prefactor is applied in a depth range "
+            "specified by the list of `Transition depths', i.e. the first "
+            "prefactor is applied above the first transition depth, the "
+            "second one between the first and second transition depth, and "
+            "so on. To compute the viscosity profile, this prefactor is "
+            "multiplied by the reference viscosity specified through the "
+            "parameter `Viscosity'. List must have one more entry than "
+            "Transition depths. Units: non-dimensional.");
+          prm.declare_entry("Minimum viscosity",
+                            "1e19",
+                            Patterns::Double(0),
+                            "The minimum viscosity cutoff. Units: Pa s");
+          prm.declare_entry("Maximum viscosity",
+                            "1e24",
+                            Patterns::Double(0),
+                            "The maximum viscosity cutoff. Units: Pa s");
+
+          // Material parameters
+          prm.declare_entry("Thermal conductivity",
+                            "4.0",
+                            Patterns::Double(0.),
+                            "Reference conductivity");
+          prm.declare_entry("Use adiabatic pressure for density",
+                            "true",
+                            Patterns::Bool(),
+                            "Whether to use adiabatic pressure instead of full "
+                            "pressure in density calculation.");
```

Comment:
This is an option we introduced in the PDA paper to allow benchmarking the PDA with the full pressure for density. But as we showed in the paper, while using the full pressure is more accurate it is also unstable. I would prefer if you removed this parameter here, and always use the adiabatic pressure for density. I do not want to propagate this option into many material models, because users might use it without understanding the implications.